packages/react/src/hooks/useVoiceAssistant.ts

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import { ParticipantKind, Track } from 'livekit-client';
+import type { RemoteParticipant } from 'livekit-client';
+import type { ReceivedTranscriptionSegment, TrackReference } from '@livekit/components-core';
+import { useRemoteParticipants } from './useRemoteParticipants';
+import { useParticipantTracks } from './useParticipantTracks';
+import { useTrackTranscription } from './useTrackTranscription';
+import { useParticipantInfo } from './useParticipantInfo';
+import { useConnectionState } from './useConnectionStatus';
+
+/**
+ * @alpha
+ */
+export type VoiceAssistantState = 'offline' | 'connecting' | 'listening' | 'thinking' | 'speaking';
+
+/**
+ * @alpha
+ */
+export interface VoiceAssistant {
+  agent: RemoteParticipant | undefined;
+  state: VoiceAssistantState;
+  audioTrack: TrackReference | undefined;
+  agentTranscriptions: ReceivedTranscriptionSegment[];
+}
+
+/**
+ * @alpha
+ *
+ * This hook looks for the first agent-participant in the room.
+ * It assumes that the agent participant is based on the LiveKit VoiceAssistant API and
+ * returns the most commonly used state vars when interacting with a VoiceAssistant.
+ */
+export function useVoiceAssistant(): VoiceAssistant {
+  const agent = useRemoteParticipants().find((p) => p.kind === ParticipantKind.AGENT);
+  const audioTrack = useParticipantTracks([Track.Source.Microphone], agent?.identity)[0];
+  const { segments: agentTranscriptions } = useTrackTranscription(audioTrack);
+  const { metadata: agentMetadata } = useParticipantInfo({ participant: agent });
+  const connectionState = useConnectionState();
+
+  const state: VoiceAssistantState = React.useMemo(() => {
+    if (connectionState === 'disconnected') {
+      return 'offline';
+    } else if (connectionState !== 'connected') {
+      return 'connecting';
+    }
+
+    if (!agent) {
+      return 'connecting';
+    }
+
+    if (!agentMetadata) {
```

Comment:
can we use participant attributes for this? `agent.state`

Reply:
Agree this would be better.
I guess ultimately the approach for state has to be decided by the `VoiceAssistant` implementation within the agents framework server side.
Right now this client implementation is mirroring the behaviour of Neil's HP agent in order to have a working prototype.

Reply:
I would advocate for changing the agent to use the new protocol that you are defining. would like to steer users into the right path for attributes vs parsing metadata

Reply:
got it, updated the PR to use attributes instead. 
This would require the agent side to set the state as `'listening' | 'thinking' | 'speaking'`. Not sure if we need anything else? (`connecting` and `disconnected` is inferred from the client's connection state and the agent's presence in the room)